interview-questions.md

```diff
@@ -14,44 +14,51 @@ Researched answer:
 
 1. What is the difference between an array's index and its value?
 
-Your answer:
+Your answer: Every element of every javascript array has an index. These indexes start at 0 and increment by 1 as elements are added to the array. These elements are the value. Arrays can hold as values; any primitive data type, other arrays, objects, and more.
 
-Researched answer:
+Researched answer: You can also have mixed type arrays. Meaning you can even have array values of more than one data type at a time. However the indexes will also be numbers starting from 0, incrementing by 1.
 
 4. Compare and contrast arrays and strings. Please include ways in which they are the same and ways in which they are different.
 
-Your answer:
+Your answer:Arrays and Strings are both collections of smaller elements. An array being made up individual values that can house all types. Strings are simply a collection of characters. Both are 0 indexed and have access to some of the same methods, such as .length.
 
-Researched answer:
+Researched answer: Arrays are considered objects, which means they can only ever be strictly equal to themselves. This is a stark contrast the primitive data type of string.
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-Your answer:
 
-Researched answer:
+Your answer:Pair programming is the process of having two people work on the same file. One programmer operates as the driver, sharing their screen and typing in the target file. The other programmer takes on the role of navigator. They simply share the mental load, assisting the driver with completion of the tasks. Driver types, Navigator researches.
+
+Researched answer: To be successful with pair programming actually invovles a high skill level from both members. The partners must have the soft skills to collaborate effectivly and stay engaged, while also having the hard skills necessary to write effective code.
 
 ## Looking Ahead: Terms for Next Week
 
 Research and define the following terms to the best of your ability.
 
-1. Higher Order Functions:
+1. Higher Order Functions: A higher order function is a function that either accepts as an argument, or returns another function. 
+
+After some time to mull over the concept, I realized that .sort is a higher order function(I think), because it accepts a function as an argument. However, I still have no meaningful understanding of how to write my own higher order functions.
+
+2. Jest: Jest is a tool used for testing javascript code. It allows you to write and run tests, and track what portions of your code have been tested and which haven't.
+
+3. Objects: An object is essentially when you hold values(varialbes) and/or behaviors(methods/functions) in its own variable.
 
-2. Jest:
+4. Method:A method in javascript, is a function that is attached to an object. 
 
-3. Objects:
+If my understanding is correct, a method is ALWAYS a function, but a function is not always a method.
```

Comment:
A method is a function, but one that is associated with an object. It is defined within the scope of an object and uses the object's data to perform actions.